### DISCUSSION TITLE
Plain TeX `$$ . . . $$` for displayed equations is discouraged in LaTeX

### DISCUSSION THREAD

**Discussion by heavywatal** [KaTeX/KaTeX#3049] [General]
According to ["Short Math Guide for LaTeX"](https://mirror.ctan.org/info/short-math-guide) published by American Mathematical Society (AMS), using `$$ ... $$` for displayed equations is strongly discouraged. Although KaTeX does not has to be consistent with AMS, it won't be very good idea to instruct beginners to get used to `$$ ... $$`. I suggest removing it from [the default set of delimiters of autorendering](https://katex.org/docs/autorender.html), or at least displaying/documenting some warnings.

> Using the plain TEX notation `$$ ... $$` for displayed equations is strongly discouraged. Although it is not expressly forbidden in LATEX, it is not documented anywhere in the LATEX book as being part of the LATEX command set, and it interferes with the proper operation of various features such as the `fleqn` option.

**Comment by edemaine** [comment]
I use `$$` all the time, and strongly disagree.  In the hundreds of math/CS paper tex files I've seen, I'm pretty sure `$$` is more common than `\[`.  Just because AMS discourages it doesn't mean that it's wrong.  (They discourage it in particular because `amsmath` doesn't get all the same spacing control with `$$`.)

Anyway, we aim for LaTeX compatibility, so given that `$$` works, it seems proper to support it by default.  (I would rather argue that `$` should be added...)

**Reply by heavywatal** [reply]
Yes, I agree it is widely used and it is not wrong. It will be almost impossible to change the existing users' habit. We will use `$$` until we die. I would admit it was too radical to suggest the removal of default support.

But how about documentation? Newer generations will eventually get familiar with `\[` if it is encouraged. Is it too annoying for the existing `$$` users?

**Reply by edemaine** [reply]
I don't think it's KaTeX's job to take any view on this, so it should just support both.  AMS has tried, but failed, to discourage use for maybe 3 decades, so I think it's pretty much impossible to change people's behavior even if we wanted to.  I personally far prefer the symmetry of `$` and `$$`, making it easy to switch between the two.
